### PR TITLE
some fixes & opt

### PR DESCRIPTION
**Pull request by mygoare**
1. fix npm package.json
2. add value field, looks better now (you can see the output changes with run `npm test`)
3. add Date filed type

### CODE REVIEW

**Comment by mygoare** [comment]
Before:

```
[2015-05-26 20:09:41] ->
{ type: 'object',
  properties: 
   { id: { type: 'number' },
     color: { type: 'string' },
     otherThings: { type: 'array', items: { type: 'string' } },
     true: { type: 'boolean' },
     false: { type: 'boolean' },
     tags: { type: 'array', items: { type: 'string' } } },
  hash: 51961 } 0ms
[2015-05-26 20:09:41] ->
{ id: 123,
  color: 'red',
  otherThings: [ 'ok', 1 ],
  true: true,
  false: false,
  tags: [ 'sun', 'warm' ] } 7ms
```

After:

```
 ✗ npm test

[2015-05-26 20:08:56] ->
{ type: 'object',
  properties: 
   { id: { type: 'number', value: 123 },
     color: { type: 'string', value: 'red' },
     otherThings: 
      { type: 'array',
        items: 
         [ { type: 'string', value: 'ok' },
           { type: 'number', value: 1 },
           { type: 'date', value: Tue May 26 2015 20:08:56 GMT+0800 (CST) } ] },
     true: { type: 'boolean', value: true },
     false: { type: 'boolean', value: false },
     tags: 
      { type: 'array',
        items: 
         [ { type: 'string', value: 'sun' },
           { type: 'string', value: 'warm' } ] } },
  hash: 10091 } 0ms
[2015-05-26 20:08:56] ->
{ id: 123,
  color: 'red',
  otherThings: [ 'ok', 1, Tue May 26 2015 20:08:56 GMT+0800 (CST) ],
  true: true,
  false: false,
  tags: [ 'sun', 'warm' ] } 8ms
```

File: src/jpack.js [hunk 1]
```diff
@@ -41,8 +41,16 @@
 
         switch (schema.type) {
         case 'array':
-            schema.items = {}
-            schemaIter(node[0], schema.items)
```
Comment (ysmood) [inline src/jpack.js:45]: No, you don't understand what json schema is. An array can only have one type of items. Something like `[1, 'str']` is illegal.

File: src/jpack.js [hunk 2]
```diff
@@ -55,6 +63,10 @@
                 schemaIter(node[key], schema.properties[key])
             }
             break
+
+        default:
+            schema.value = node;
```
Comment (ysmood) [inline src/jpack.js:68]: A schema should never have instance value inside.

File: package.json [hunk 3]
```diff
@@ -2,9 +2,9 @@
   "name": "jpack",
   "version": "0.0.1",
   "description": "",
-  "main": "main.js",
+  "main": "src/jpack.js",
   "scripts": {
-    "test": "npm test"
+    "test": "node ./test/lab.js"
```
Comment (ysmood) [inline package.json:7]: I won't accept you change, I use nokit to test. See my project `package.json` like https://github.com/ysmood/yaku

**Comment by ysmood** [comment]
This project is on its very early stage. Don't make PR for it. Do it when it has a readme.

**Comment by mygoare** [comment]
It seems I misunderstand your repository.Sorry about that, my fault. I understand what JSON Schema is now.

**Pull request closed without merging**